pkg/bootstrap/cert.go

```diff
@@ -126,17 +126,20 @@ func CertificateFromRaw(rawCert, rawKey []byte) (*Certificate, error) {
 }
 
 // ReadCertificates read certificate and key from folder.
-func ReadCertificates(dir string) (*Certificate, error) {
+func ReadCertificates(dir string, withKey bool) (*Certificate, error) {
 	// Read certificate
 	rawCert, err := os.ReadFile(filepath.Join(dir, config.CertificateFileName))
 	if err != nil {
 		return nil, err
 	}
 
-	// Read key
-	rawFabricKey, err := os.ReadFile(filepath.Join(dir, config.PrivateKeyFileName))
-	if err != nil {
-		return nil, err
+	var rawFabricKey []byte
+	if withKey {
+		// Read key
+		rawFabricKey, err = os.ReadFile(filepath.Join(dir, config.PrivateKeyFileName))
```

Comment:
Not sure if OS.ReadFile can return nil without an error, if it can maybe it is better to propagate also the boolean "withkey" to handle this case.